### PR TITLE
Fix XML load CPU hotspot: Scrub.c xaccTransScrubPostedDate

### PR DESCRIPTION
**Pull request by chriscarson60187**
The refactoring provides roughly 10% reduction in user CPU use for XML file load by moving an expensive function to within an if-clause where the result is used.  The diff look like a full re-write of the function but only the if statements, indenting, and commentary changed.

Disclosure #1:  My first attempt at the pull request mechanism
Disclosure #2: I rebased my branch off of upstream/maint right before submitting this... and tied a holy knot in my own repository, but I don't think it impacts this at all.  Gah.  Learning.

### CODE REVIEW

**Comment by jralls** [comment]
Good catch.

File: libgnucash/engine/Scrub.c [hunk 1]
```diff
@@ -1394,14 +1394,30 @@ xaccScrubUtilityGetOrMakeAccount (Account *root, gnc_commodity * currency,
...
+    // 10% reduction user space CPU use during
+    // XML file load.
+    //
+    // The original prepared the alternate date, i.e.,
+    // GDate date = xaccTransGetDate....
+    // before checking whether it was needed.
+    // Surprise, calculating the alternate date
+    // is expensive and is rarely needed,
+    // so the reordering is a win.
+
```
[the remark this answers is not in the document]
Comment (chriscarson60187) [reply libgnucash/engine/Scrub.c]: Added a commit to remove the comment.
ed7ea90052a0c2636024c07a91cb95f8ab78292a

**Comment by jralls** [comment]
You still need to clean up the history in this branch so that it's a single commit.

**Comment by chriscarson60187** [comment]
Single commit pushed

**Comment by jralls** [comment]
Thanks!

**Pull request merged**